daemon/src/engine/qcommon/q_shared.cpp

```diff
@@ -1868,6 +1868,61 @@ char     *QDECL PRINTF_LIKE(1) va( const char *format, ... )
 =====================================================================
 */
 
+std::string InfoMapToString( const InfoMap& map )
+{
+	std::string info_string;
+
+	for ( const auto& pair : map )
+	{
+		if ( pair.first.find( INFO_SEPARATOR ) != std::string::npos ||
+			pair.second.find( INFO_SEPARATOR ) != std::string::npos )
+		{
+			Log::Notice( "Can't use keys or values with a \\\n" );
+			continue;
+		}
+
+		if ( pair.second.empty() )
+		{
+			continue;
+		}
+
+		info_string += INFO_SEPARATOR + pair.first + INFO_SEPARATOR + pair.second;
+	}
+
+	return info_string;
+}
+
+
+InfoMap InfoStringToMap( const std::string& string )
+{
+	if ( string.empty() )
+	{
+		return InfoMap{};
+	}
+
+	InfoMap map;
+	std::istringstream input( string );
+	input.ignore( 1, INFO_SEPARATOR );
+	while ( input.good() )
```

Comment:
I'm trying to deprecate those old Info functions, they are a bit ugly and potentially unsafe.
Info_NextPair for instance has no bound checking and makes a few assumptions on the pointers it receives without documenting them.